cmd/kafka-consumer/writer.go

```diff
@@ -384,13 +384,22 @@ func (w *writer) WriteMessage(ctx context.Context, message *kafka.Message) bool
 			watermark := atomic.LoadUint64(&progress.watermark)
 			// if the kafka cluster is normal, this should not hit.
 			// else if the cluster is abnormal, the consumer may consume old message, then cause the watermark fallback.
-			if row.CommitTs < watermark {
-				log.Panic("RowChangedEvent fallback row, ignore it",
+			if row.CommitTs < watermark && message.TopicPartition.Offset > progress.watermarkOffset {
+				if message.TopicPartition.Offset > progress.watermarkOffset {
+					log.Panic("RowChangedEvent fallback row, ignore it",
```

Comment:
IMO, we panic here since the offset of single partition should never fall back?  If so, please correct the log message.